### PR TITLE
Add Chinese translation

### PR DESCRIPTION
**Pull request by ZeroCiel**
I don't know much about GitHub.
I tried to add Chinese content under loc.
But I still find some English is missing, please try to sort them if you can.

I want to leave an address in the file that can be used to contact me if the player wants to correct an incorrect translation.

It's in line 896.
If you don't think it's appropriate you can delete the first piece of code.

If you have any other questions, please contact me.

### CODE REVIEW

File: loc/CN/strings_db.lua [hunk 1]
```diff
@@ -0,0 +1,8710 @@
...
+MP_Taunt_0156="圣者承诺将赐予跟随他的人荣耀的赏赐。"
+MP_Taunt_0157="圣者是令UEF部队闻风丧胆。"
+MP_Taunt_0158="只有圣者马可松能够保护我们免于遭到「黑日」的荼毒。"
+MP_Taunt_0159="你应该和我并肩作战，而不是与我为敌！"
+MP_Taunt_0160="沒有人能阻止UEF！"
+MP_Taunt_0161="我不会让你们毁灭这个星球。"
+MP_Taunt_0162="「万古」的部队是由凶手和疯子组成。"
+MP_Taunt_0163="这就是強大的「万古启示」？一点都不能让我感觉印象深刻。"
+MP_Taunt_0164="我想你低估我了。"
+MP_Taunt_0165="我绝对
```
Comment (shalkya) [inline loc/CN/strings_db.lua]: luac: ./loc/CN/strings_db.lua:8325: invalid escape sequence near '\�'
you have \n in here, i don't think it's supposed to be here.

**Comment by ZeroCiel** [comment]
I tried to check it, but I couldn't find \n or \，So I tried to translate it in a different way, can you still see that byte?

File: loc/CN/strings_db.lua (hunk 1 above)
Comment (ZeroCiel) [reply loc/CN/strings_db.lua]: My vscode shows normal, I have modified the translation once, if there is still this problem, try to inform me again.

**Comment by shalkya** [comment]
smh it shown the wrong line during my review, very weird i'm redoing it

**Comment by ZeroCiel** [comment]
> smh it shown the wrong line during my review, very weird i'm redoing it
> it's this line that you need to update : https://github.com/FAForever/fa/pull/3015/files/83d4cddd5477145d0c6e4c730e6c68e3a90e576f#diff-0fb4cb8cc6fed19a84c75fb875fdc934R8325

ok.I fixed them.

**Comment by KionX** [comment]
tooltipui0714, tooltipui0715 - missing new lines.

tooltipui0134 - wrong characters 「」
tooltipui0636 - wrong characters ： use this :

**Comment by ZeroCiel** [comment]
Yes, I see it. We can't use \n here？at right?
Most of the text has been modified since the traditional version,

If I can't find the new rows in the US version, then I can't translate them.
Because I don't know exactly where the code is.
（About this part of the code:  tooltipui0714, tooltipui0715）

[Tooltipui0636]I am not sure if it is correct, please check again.
If the format is wrong, please tell me the correct format.

**Comment by ZeroCiel** [comment]
I noticed that the names of some units were missing, but I couldn't find their English in the US version.
(T3 MAA and new factory)

**Comment by KionX** [comment]
tooltipui0134, Tooltipui0636 - ok.

tooltipui0714, tooltipui0715 here
https://github.com/FAForever/fa/blob/deploy/fafdevelop/loc/US/strings_db.lua
deploy/fafdevelop the actual development branch.

**Comment by germanicianus** [comment]
> I tried to add Chinese content under loc.

I had a quick look. Why do we need (simplified) Chinese? To my knowledge people from mainland China are able to read traditional Chinese which is already available in the TW (Taiwan) directory. We could throw the TW loc file into a converter like https://www.lexilogos.com/keyboard/chinese_conversion.htm and get the conversion for free. There may be some minor differences in grammar, but I think it would be sufficient. Correct me if I'm wrong.

**Comment by ZeroCiel** [comment]
> I had a quick look. Why do we need (simplified) Chinese? To my knowledge people from mainland China are able to read traditional Chinese which is already available in the TW (Taiwan) directory. We could throw the TW loc file into a converter like https://www.lexilogos.com/keyboard/chinese_conversion.htm and get the conversion for free. There may be some minor differences in grammar, but I think it would be sufficient. Correct me if I'm wrong.

In fact, steam's Chinese players have been using simplified Chinese all along.
But the reality is that most people don't actively read foreign news.

And, as you say,We do have a simplified document here that is converted from traditional Chinese.
It's what you said it was,But it still has many mistakes.

I also proofread the TW text carefully.It's a question of work attitude.
It's not appropriate to call it CN.
There is a difference between linguistic logic.
I think I can give you a lot of examples, but it's hard for me to explain.
Even with the help of translation software to translate them.

This is a problem I often encounter when translating English into Chinese.
Some English is too simple and cannot be translated into Chinese to express the correct meaning.

It is logically correct in English.
But in Chinese it's a different word.

like：desync_0003="Game state has become inconsistent between the players in this game. This means the simulation may evolve differently for different players from this point. The game is extremely likely to terminate abruptly quite soon."

If you use translation software to translate it
Then it will become:"游戏状态在玩家之间变得不一致。这意味着从这一点开始，不同玩家的模拟可能会有不同的发展。这场游戏极有可能很快就会突然终止。"

But I wrote it this way.
desync_0003="游戏版本不一致，请检查双方版本号是否一致，如果继续极有可能导致游戏崩溃。"

Translated into English, it is understood this way.
Desync_0003 =" the game version is not consistent, please check whether the two versions are consistent, if continue to cause the game crash."

Or you can compare my files with TW.
Even if it (TW) uses Microsoft word to convert to simplified Chinese.
You wouldn't know the difference.

Like this:
CN:玩家 TW:玩者 CN:网络 TW:网路 CN:纳米 TW:奈米

Of course, the TW files also have a lot of errors, they spell the words wrong.
When I proofread the translation, I saw these mistakes.
But I don't think that's my job.

English, traditional Chinese, simplified Chinese, the three languages have different grammatical logic.

At present, there are parts of English that I still cannot translate, there are too many differences between the two.

I may leave it alone, or until I understand what it means.

One more thing.
As far as I know, fewer than a few thousand Chinese players know about FAF.
Yes, but most users choose to launch their games offline.
One of the reasons is because the server experience is so bad.
So they like to create LAN to play games.
So FAF has very few Chinese players online.

Another reason is that due to the influence of Great Wall, it takes a long time to download patches using FAF platform.If you can solve the problem of downloading files,Then there will be more Chinese players willing to use FAF.
(The Internet here is really terrible.)

**Comment by ZeroCiel** [comment]
> tooltipui0134, Tooltipui0636 - ok.
> tooltipui0714, tooltipui0715 here
> https://github.com/FAForever/fa/blob/deploy/fafdevelop/loc/US/strings_db.lua
> deploy/fafdevelop the actual development branch.

I adjusted the sequence of code between 670 and 715,
I will continue to translate when I have free time.
I need some time to think about them.

**Comment by ZeroCiel** [comment]
HMM... Can't I change their order?

**Comment by germanicianus** [comment]
Please don‘t mix up the terms when speaking about Chinese characters (simplified/traditional) and Chinese „regional“ dialects (e. g. mainland/Cantonese/Taiwanese - but not regarding pronunciation).

I am well aware that automatic translation (not conversion) is crappy and cannot be used.

What I would like to see is a cleaned up/corrected TW loc file holding modern standard Mandarin Chinese using traditional characters. This would be the single point of truth for Chinese and eliminate the need to maintain two files.

**Comment by germanicianus** [comment]
I think you also should only correct the FAF translation parts but leave the original SC:FA TW part as is because that is the official one of the publisher.

**Comment by ZeroCiel** [comment]
It's a messy problem, to be honest.

If you think the game story text needs to be preserved, 
I can replace it with the text back.

Some users prefer traditional characters, while others are more comfortable with simplified characters.
Write the TW and CN files together, and the results will be terrible.
CN and TW have different logical syntax,
There are quite a few words that will confuse china mainland users.

If you insist on using the traditional Chinese characters game interface,Then mainland Chinese users are unlikely to join in, because compared to what you said,They have better options.

A long time ago, some Chinese people made a translation of the text. They converted traditional Chinese into simplified Chinese.

But if you change the interface to simplified Chinese, you will lose TW and HK users, even if there are many errors in the translated text.
But if you choose traditional Chinese, you still don't get mainland Chinese users.

They will use the files of the translation group I mentioned.

Their file is full of advertising, spam and lots of errors. I don't recommend users use this translation file, but they do it anyway.

As I said before, this is a terrible question.Some of TW and HK prefer to use traditional Chinese.Mainland Chinese users prefer simplified Chinese.

Whichever option you choose,it's not a good idea.
If you're willing to respect your users' habits,Two different versions of the file are necessary.

**Comment by speed2CZ** [comment]
Just because its original, it doesnt mean its good. If it needs corrections, do it.

**Comment by ZeroCiel** [comment]
In the last update, I seem to have pressed an extra space so that the translation cannot be recognized.I couldn't find it, so I used another backup file to upload.

The character name code is [{I name}] instead of [{iname}]I confused the problem by clearing Spaces.
so I may have to fix it again later.Otherwise the player will not see the correct character name in the cooperative campaign.

In fact, in the original story text, there were a lot of spacebars.
There are always unnecessary space symbols when players read the story of the game.

So they should be removed.I haven't changed the story text much, just corrected the language logic just corrected its language logic so that every player can understand it.

**Comment by ZeroCiel** [comment]
If you want it to support traditional Chinese, just convert it back to traditional Chinese and you can use it.

Only a portion of the Chinese byte characters need to be replaced.
"玩家"="玩者"
"网址"="网路"
"纳米"="纳米"
"防御墙"="墙区段"
But the bad thing is I can't do that.
When I used word to convert text, the game couldn't recognize the Chinese characters.
So I always proofread the text manually.

If you want to support TW, you need a better way to convert them.
At least for me, I won't.

**Comment by ZeroCiel** [comment]
8485~8640 I'm not sure about this part, I'm not a PVP player so I don't know much about this part of the setup.
8763~8831 This part of the translation is from an, and I have rewritten part of his translation.
If this part of the text is missing, then part of the player's unit description will be in English.
8377~8478 More detailed configuration instructions.
8098~8266 Sorted the scrambled text, but I couldn't find any other English text.

Some of the things I'm not so sure about are not translated.
Some grammar is beyond my ability.Or some words I can't find in the dictionary
Or maybe I don't know how to explain it exactly.

I also noticed a BUG.
[Unit_Description_0316]It's actually a wrong translation, but I can't find the English original.
It's actually a description of a satellite weapon.It's not engineering.

**Comment by KionX** [comment]
Not all text is in _db.lua

**Comment by ZeroCiel** [comment]
Can it be destroyed by a nuclear bomb?
I've heard players say that if you send a nuclear bomb to the current location of a satellite,
Then can be destroyed.

I haven't tried it in the game. Do you know that?

**Comment by ZeroCiel** [comment]
Excuse me, Do you know where to view the dialogue file of AI?
Information text like XGG_GP1 and XGG_MP1.（like 3692 to 3723 line）

I want to translate the information the AI sends to the player.
If it's convenient for you,Please tell me.

Thank you very much!

**Comment by KionX** [comment]
https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/ui/game/taunt.lua

This cannot be localized:
https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/AI/sorianlang.lua#L230

I don't know about satellite.

**Comment by shalkya** [comment]
a satellite will die to a nuke (before it wouldn't die, and we changed it few years ago)
i can't review chinese, and germanicianus seemed to have brief look at it.
If you still have question you can ask them.
Tell us when you are satisfied of your change. I will merge your PR at that point.

**Comment by ZeroCiel** [comment]
To be honest, I don't know how to write this code.（Lines 10 to 322 for sorianlang.lua）
I guess it's a "AIChatText_nukechat_001" format,
But I was wrong.So I would like to ask how to translate this part of the code?

If possible, please provide their format. Thank you very much!

**Comment by KionX** [comment]
It important? It can't be localized right now. These lines are for SorianAI.
Instead of Sorian AI i recommend using AI-Uveso.

**Comment by KionX** [comment]
Missing CN text: [Report_CN.txt](https://github.com/FAForever/fa/files/4372644/Report_CN.txt)

**Comment by ZeroCiel** [comment]
I'll take care of it tonight

**Comment by ZeroCiel** [comment]
I have uploaded it, because there was a little problem in the test.
So I updated it twice

**Comment by ZeroCiel** [comment]
Do we have that?I found some duplicate Chinese text in the unit management Settings.(In the game)

they like [restricted_units_info_DIRECTFIRE_BASE_T1] and [restricted_units_data_DIRECTFIRE_BASE_T1]

I guess they are.... 
restricted_units_info_DIRECTFIRE_BASE_T2 and restricted_units_data_DIRECTFIRE_BASE_T2
or restricted_units_info_DIRECTFIRE_BASE_T3 and restricted_units_data_DIRECTFIRE_BASE_T3

Was my guess right?

**Comment by KionX** [comment]
I think so too. You can add these lines in advance.
Need an additional fix to enable these lines. I'll take care of it.

**Comment by ZeroCiel** [comment]
You can merge them.There's no need to change the rest.
There are some parts of English that I cannot translate accurately.
I don't know much about English abbreviations.
It's probably beyond my capacity.

If you have any other questions, please let me know.

**Comment by shalkya** [comment]
found some english, so there still room for improvement.
But you did a huge job of translation.

**Pull request merged**